io/io.go

```diff
@@ -82,6 +82,17 @@ type WriteFileIO interface {
 	WriteFile(name string, p []byte) error
 }
 
+// ListIO is the interface implemented by a file system that
+// provides listing functionality for object stores
+type ListIO interface {
+	IO
+
+	// ListObjects lists all objects with the given prefix.
+	// For object stores like S3, this provides proper prefix-based listing.
+	// The callback function is called for each object found.
+	ListObjects(prefix string, fn func(path string, info fs.FileInfo) error) error
```

Comment:
could we just use https://pkg.go.dev/io/fs#WalkDir ?

`BlobFS` implements https://pkg.go.dev/io/fs#Sub which we can use